src/DocBlock/Annotation.php

```diff
@@ -16,10 +16,46 @@
  * This represents an entire annotation from a docblock.
  *
  * @author Graham Campbell <graham@alt-three.com>
+ * @author Dariusz Rumiński <dariusz.ruminski@gmail.com>
  */
 class Annotation
 {
     /**
+     * Regex to match any types, shall be used with `x` modifier.
+     *
+     * @internal
+     */
+    const REGEX_TYPES = '
+    # <scalar> is any non-array, non-generic, non-alternated type, eg `int` or `\Foo`
```

Comment:
:+1: for simple, does this apply to simple values as well like `false` in `@return false|array<int, false>` ?